### PR TITLE
feat: introduce props for emptyViewheight and the emptyviewButton

### PR DESCRIPTION
**Pull request by YasinduRC**
### Summary

-

### How to test

1.

### Project Checklist

- [ ] Changes build without any errors
- [ ] Have written adequate test cases
- [ ] Done developer testing in
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
- [ ] Code is formatted with `npm run format`
- [ ] Code is linted with `npm run check-lint`
- [ ] No unnecessary comments left in code
- [ ] Made corresponding changes to the documentation

#### Other

- [ ] New atomic components added
- [ ] New molecules added
- [ ] New pages(routes) added
- [ ] New dependencies installed

### PR Checklist

- [ ] Pull request is raised from the correct source branch
- [ ] Pull request is raised to the correct destination branch
- [ ] Pull request is raised with correct title
- [ ] Pull request is self reviewed
- [ ] Pull request is self assigned
- [ ] Suitable pull request status labels are added (`ready-for-code-review`)

### Additional Information
-

### CODE REVIEW

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

This PR adds two optional props (`emptyViewHeight` and `showEmptyViewButton`) to `SidePanelDealSection`, wrapping the empty-state `EmptyDataView` in a height-controlled div and making its action button conditional. Behavior is preserved: the default `14.25rem` equals the previous `228px` and the button still renders by default. The change is clean and type-safe; findings are minor.

Found **5** new issue(s):

File: frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx [hunk 1]
```diff
@@ -59,21 +65,27 @@ const SidePanelDealSection: React.FC<Props> = ({ deals }) => {
...
-            variant: "tertiary",
-            onClick: handleAddDeal,
-            icon: <PlusIcon />,
-            "aria-label": translateText(["ariaLabels", "addDealBtn"])
-          }}
-          className={{
-            wrapper: "h-[228px] bg-secondary-background rounded-lg"
-          }}
-        />
+        <div style={{ height: emptyViewHeight }}>
```
Comment (ThinuwanW) [inline frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx]: 🤖 **Claude** · 💬 **Note**

Inline `style={{ height: emptyViewHeight }}` is used here. Per the styling guidelines, inline styles are generally discouraged in favor of Tailwind classes / colocated styles. Since this is a runtime prop value it can't be a static Tailwind arbitrary class, so an inline style is a reasonable pragmatic choice — but consider using a CSS custom property (e.g. `style={{ ['--empty-h']: emptyViewHeight }}` with a `h-[var(--empty-h)]` class) for consistency with the surrounding Tailwind-based styling. Nit — low confidence; dev's call.

File: frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx [hunk 2]
```diff
@@ -17,9 +17,15 @@ import DealAccordionItemHeader from "./DealAccordionItemHeader";
 
 interface Props {
   deals: CrmDealType[];
+  emptyViewHeight?: string;
```
Comment (ThinuwanW) [inline frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx]: 🤖 **Claude** · 💬 **Note**

The two new props `emptyViewHeight` and `showEmptyViewButton` are added but not consumed by either caller (`CompanySidePanel.tsx:79` and `ContactSidePanel.tsx:119` both pass only `deals`). If the intent was to give these panels a different empty-view height or hide the add button, that wiring appears to be missing from this PR. If this is intentional groundwork for a follow-up, it's fine to leave as-is; otherwise wire the props at the call sites so the new configurability is actually exercised.

File: frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx [hunk 3]
```diff
@@ -17,9 +17,15 @@ import DealAccordionItemHeader from "./DealAccordionItemHeader";
...
+  showEmptyViewButton?: boolean;
```
Comment (ThinuwanW) [inline frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx]: 🤖 **Claude** · 💬 **Note**

The new `showEmptyViewButton` prop only gates the Add Deal button in the empty-state branch. The populated (`hasDeals`) branch still renders its own Add Deal `ButtonV2` (around lines 46-57) unconditionally. If a caller passes `showEmptyViewButton={false}` to obtain a read-only/non-actionable deal section, the add-deal affordance will still appear whenever deals exist, defeating the intent. Consider gating the populated-state button with the same flag (or renaming the prop to make it explicit that it only controls the empty view, e.g. leave as-is but document the scope).

File: frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx [hunk 4]
```diff
@@ -17,9 +17,15 @@ import DealAccordionItemHeader from "./DealAccordionItemHeader";
...
 }
 
-const SidePanelDealSection: React.FC<Props> = ({ deals }) => {
+const SidePanelDealSection: React.FC<Props> = ({
+  deals,
+  emptyViewHeight = "14.25rem",
```
Comment (ThinuwanW) [inline frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx]: 🤖 **Claude** · 💬 **Note**

The default `emptyViewHeight = "14.25rem"` is a magic dimension chosen to reproduce the previous `h-[228px]` (228px = 14.25rem). This ties the design system's spacing to an opaque string literal. Consider extracting it to a named constant with a comment noting it equals the prior 228px, so the intent is discoverable and the value isn't silently drifted in future edits.

File: frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx (hunk 2 above)
Comment (ThinuwanW) [inline frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx]: 🤖 **Claude** · 💬 **Note**

📁 **File-level observation**

No `SidePanelDealSection.test.tsx` exists, and this PR introduces new conditional rendering (button shown vs. `undefined` based on `showEmptyViewButton`, and custom `emptyViewHeight`). Per the testing guidelines, add tests covering: (1) empty state renders the button when `showEmptyViewButton` defaults to true, (2) button is absent when `showEmptyViewButton={false}`, and (3) the wrapper height reflects a custom `emptyViewHeight`. Use the `MockTheme` wrapper and prefer `screen.getByRole`.

File: frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx (hunk 1 above)
Comment (anusham-rc) [reply frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx]: Resolved: prop removed; empty view now uses a static h-[228px] class.

File: frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx (hunk 2 above)
Comment (anusham-rc) [reply frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx]: Resolved: replaced by showAddDealAction/emptyDescription, both wired in TaskSidePanel.

File: frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx (hunk 3 above)
Comment (anusham-rc) [reply frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx]: Resolved: showAddDealAction now gates the list button, inline form, and empty-view button.

File: frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx (hunk 4 above)
Comment (anusham-rc) [reply frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx]: Resolved: default prop removed, height is a static Tailwind class again.

File: frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx (hunk 2 above)
Comment (anusham-rc) [reply frontend/src/community/crm/components/molecules/SidePanelDealSection/SidePanelDealSection.tsx]: Valid: no test files for the new conditional rendering yet - will follow up.

**Review by akilarootcode — APPROVED** [review]

**Review by thusala — APPROVED** [review]

**Review by ThinuwanW — APPROVED** [review]
